sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala

```diff
@@ -89,13 +89,7 @@ private[sql] object CatalogV2Implicits {
   }
 
   implicit class IdentifierHelper(ident: Identifier) {
-    def quoted: String = {
-      if (ident.namespace.nonEmpty) {
-        ident.namespace.map(quote).mkString(".") + "." + quote(ident.name)
-      } else {
-        quote(ident.name)
-      }
-    }
+    def quoted: String = ident.toString
```

Comment:
Identifier is an interface, so we can't rely on its `toString` method to have predictable behavior. This should use the original implementation.